online-data-tools/fetch_gowdy_supplement.py

```
@@ -0,0 +1,155 @@
+    r"<a\s+href=['\"]/read/UD/[0-9a-fA-F]{4}/([0-9a-fA-F]{4})['\"]>\s*[0-9a-fA-F]{4}\s*</a>\s*<td>([^<]*)",
+    re.IGNORECASE,
+)
+
+
+def _fetch(url: str, *, timeout: float = 30.0) -> str:
+    # gowdy.us serves a self-signed-ish cert behind sourceforge — accept it.
+    # The scraped content is structural / public; SSL-stripping risk is low.
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    req = urllib.request.Request(url, headers={"User-Agent": "fbuild-bot/1.0"})
```

Ruff (0.15.17)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)